.github/workflows/nix.yml

```diff
@@ -41,14 +43,43 @@ jobs:
         extra_nix_config: |
           experimental-features = nix-command flakes
         nix_path: nixpkgs=channel:nixos-unstable
-    - if: ${{ needs.pre_job.outputs.should_skip != 'true' && env.HAS_TOKEN == 'true' }}
-      uses: cachix/cachix-action@v8
+    - if: ${{ needs.pre_job.outputs.should_skip != 'true' }}
+      uses: cachix/cachix-action@v10
       with:
         name: haskell-language-server
-        authToken: ${{ secrets.HLS_CACHIX_AUTH_TOKEN }}
+        # Disable pushing, we will do that in job `build`
+        skipPush: true
     - if: ${{ needs.pre_job.outputs.should_skip != 'true' }}
-      run: nix build
-    - if: ${{ needs.pre_job.outputs.should_skip != 'true' && env.HAS_TOKEN == 'true' }}
-      run: nix develop --profile dev && cachix push haskell-language-server dev
-    - if: ${{ needs.pre_job.outputs.should_skip != 'true' && env.HAS_TOKEN == 'true' }}
-      run: nix path-info --json | jq -r '.[].path' | cachix push haskell-language-server
+      run: |
+        nix develop --command cabal update
+        nix develop --command cabal build
+
+  # Build and then push HLS binaries with developmet shell to cachix
+  # This job runs when PRs are merged to master, and should be excluded from branch protections
+  build:
+    runs-on: ${{ matrix.os }}
+    if: ${{ github.repository_owner == 'haskell' && github.ref == 'ref/heads/master' }}
```

Comment:
It didn't run as expected :(

https://github.com/haskell/haskell-language-server/actions/runs/870310690